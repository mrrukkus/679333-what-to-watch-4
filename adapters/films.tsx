const createFilm = (data) => {
  return {
    runTime: data.run_time,
    img: data.preview_image,
    title: data.name,
    genre: data.genre,
    year: data.released,
    filmBackground: data.background_image,
    poster: data.poster_image,
    score: data.rating,
    ratingCount: data.scores_count,
    paragraphs: data.description,
    director: data.director,
    starring: data.starring,
    preview: data.preview_video_link,
    isFavorite: data.is_favorite,
    video: data.video_link,
    id: data.id,
  };
};

interface Film {
  runTime: number,
  img: string,
  title: string,
  genre: string,
  year: number,
  filmBackground: string,
  poster: string,
  score: number,
  ratingCount: number,
  paragraphs: string,
  director: string,
  starring: string[],
  preview: string,
  isFavorite: boolean,
  video: string,
  id: number
  comments: any
}

export {Film};
export default createFilm;
