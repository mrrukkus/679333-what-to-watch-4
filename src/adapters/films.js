const createFilm = (data) => {
  return {
    runTime: data.run_time,
    img: data.preview_image,
    title: data.name,
    genre: data.genre,
    year: data.released,
    filmBackground: data.background_image,
    poster: data.poster_image,
    score: data.rating,
    ratingCount: data.scores_count,
    paragraphs: data.description,
    director: data.director,
    starring: data.starring,
    preview: data.video_link,
    isFavorite: data.is_favorite,
    id: data.id
  };
};

export default createFilm;
