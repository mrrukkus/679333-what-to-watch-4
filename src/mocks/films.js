export default [
  {
    runTime: `1h 39m`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Drama`,
    year: 2014,
    filmBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    score: `1,0`,
    ratingLevel: `Very good`,
    ratingCount: 240,
    paragraphs: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`, `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    runTime: `1h 39m`,
    img: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`,
    genre: `Drama`,
    year: 2014,
    filmBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    score: `2,0`,
    ratingLevel: `Very good`,
    ratingCount: 240,
    paragraphs: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`, `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    runTime: `1h 39m`,
    img: `img/macbeth.jpg`,
    title: `Macbeth`,
    genre: `Drama`,
    year: 2014,
    filmBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    score: `8,9`,
    ratingLevel: `Very good`,
    ratingCount: 240,
    paragraphs: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`, `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    runTime: `1h 39m`,
    img: `img/aviator.jpg`,
    title: `Aviator`,
    genre: `Drama`,
    year: 2014,
    filmBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    score: `3,0`,
    ratingLevel: `Very good`,
    ratingCount: 240,
    paragraphs: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`, `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    runTime: `1h 39m`,
    img: `img/we-need-to-talk-about-kevin.jpg`,
    title: `We need to talk about Kevin`,
    genre: `Noir`,
    year: 2014,
    filmBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    score: `4,9`,
    ratingLevel: `Very good`,
    ratingCount: 240,
    paragraphs: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`, `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    runTime: `1h 39m`,
    img: `img/what-we-do-in-the-shadows.jpg`,
    title: `What We Do in the Shadows`,
    genre: `Noir`,
    year: 2014,
    filmBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    score: `5,1`,
    ratingLevel: `Very good`,
    ratingCount: 240,
    paragraphs: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`, `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    runTime: `1h 39m`,
    img: `img/revenant.jpg`,
    title: `Revenant`,
    genre: `Noir`,
    year: 2014,
    filmBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    score: `7,0`,
    ratingLevel: `Very good`,
    ratingCount: 240,
    paragraphs: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`, `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    runTime: `1h 39m`,
    img: `img/johnny-english.jpg`,
    title: `Johnny English`,
    genre: `Noir`,
    year: 2014,
    filmBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    score: `37,9`,
    ratingLevel: `Very good`,
    ratingCount: 240,
    paragraphs: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`, `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }
];
