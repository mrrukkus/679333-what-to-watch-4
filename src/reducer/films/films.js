import {extend, DEFAULT_CARDS_COUNT, MORE_LIKE_THIS_CARDS_COUNT} from "../../utils.js";

const getFilteredFilmsList = (genre, filmsList) => {
  if (genre !== `All genres`) {
    return filmsList.filter((film) => film.genre === genre);
  }

  return filmsList;
};

const calculateNewCardsCount = (currentCardsCount, increaser, filmsCount) => {
  let newCardsCount = currentCardsCount + increaser;

  if (filmsCount < newCardsCount) {
    newCardsCount = currentCardsCount + (filmsCount - currentCardsCount);

    return newCardsCount;
  }

  return newCardsCount;
};

const initialState = {
  genre: `All genres`,
  filmToRenderDetails: null,
  filmToPlay: null,
  currentFilmsCardsCount: DEFAULT_CARDS_COUNT,
  postCommentStatus: null,
};

const ActionType = {
  FILTER_CHANGE: `FILTER_CHANGE`,
  SHOW_DETAILS: `SHOW_DETAILS`,
  CHANGE_CARDS_COUNT: `CHANGE_CARDS_COUNT`,
  PLAY_FILM: `PLAY_FILM`,
  EXIT_FILM: `EXIT_FILM`,
  POST_COMMENT: `POST_COMMENT`
};

const ActionCreator = {
  filterChange: (genreType) => ({
    type: ActionType.FILTER_CHANGE,
    genre: genreType,
  }),

  showDetails: (film) => ({
    type: ActionType.SHOW_DETAILS,
    filmToRenderDetails: film
  }),

  changeCardsCount: (count) => ({
    type: ActionType.CHANGE_CARDS_COUNT,
    currentFilmsCardsCount: count
  }),

  playFilm: (film) => ({
    type: ActionType.PLAY_FILM,
    filmToPlay: film
  }),

  exitFilm: () => ({
    type: ActionType.EXIT_FILM,
    filmToPlay: null
  }),

  postComment: (status) => ({
    type: ActionType.POST_COMMENT,
    postCommentStatus: status
  })
};

const Operation = {
  postComment: (commentData) => (dispatch, getState, api) => {
    return api.post(`/comments/${commentData.id}`, {
      "rating": commentData.rating,
      "comment": commentData.comment,
    })
      .then((response) => {
        dispatch(ActionCreator.postComment(response.status));
      });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.FILTER_CHANGE:
      return extend(state, {
        genre: action.genre
      });
    case ActionType.SHOW_DETAILS:
      return extend(state, {
        filmToRenderDetails: action.filmToRenderDetails,
        currentFilmsCardsCount: MORE_LIKE_THIS_CARDS_COUNT
      });
    case ActionType.CHANGE_CARDS_COUNT:
      return extend(state, {
        currentFilmsCardsCount: action.currentFilmsCardsCount
      });
    case ActionType.PLAY_FILM:
      return extend(state, {
        filmToPlay: action.filmToPlay,
      });
    case ActionType.EXIT_FILM:
      return extend(state, {
        filmToPlay: action.exitFilm
      });
    case ActionType.POST_COMMENT:
      return extend(state, {
        postCommentStatus: action.postCommentStatus
      });
  }
  return state;
};

export {reducer, calculateNewCardsCount, getFilteredFilmsList, ActionType, ActionCreator, Operation};
