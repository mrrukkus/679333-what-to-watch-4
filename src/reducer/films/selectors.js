import NameSpace from "../name-space.js";

const NAME_SPACE = NameSpace.FILMS;

export const getGenre = (state) => {
  return state[NAME_SPACE].genre;
};

export const getFilmToRenderDetails = (state) => {
  return state[NAME_SPACE].filmToRenderDetails;
};

export const getFilmToPlay = (state) => {
  return state[NAME_SPACE].filmToPlay;
};

export const getCurrentFilmsCardsCount = (state) => {
  return state[NAME_SPACE].currentFilmsCardsCount;
};

