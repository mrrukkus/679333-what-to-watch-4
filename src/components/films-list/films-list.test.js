import React from "react";
import renderer from "react-test-renderer";
import films from "../../mocks/films.js";
import FilmsList from "../films-list/films-list.jsx";

it(`Films list renders correctly`, () => {
  const list = renderer
    .create(<FilmsList
      filmsList={films.filmsForCards}
      onCardAction={() => {}}
      openDetailsHandler={() => {}}
    />).toJSON();

  expect(list).toMatchSnapshot();
});
