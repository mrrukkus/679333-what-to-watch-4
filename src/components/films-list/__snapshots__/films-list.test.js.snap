// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films list renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <a
    className="small-movie-card catalog__movies-card small-movie-card__link"
    href="/films/0"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <span
        className="small-movie-card__link"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </span>
    </h3>
  </a>
  <a
    className="small-movie-card catalog__movies-card small-movie-card__link"
    href="/films/1"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Bohemian Rhapsody"
        height="175"
        src="img/bohemian-rhapsody.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <span
        className="small-movie-card__link"
        onClick={[Function]}
      >
        Bohemian Rhapsody
      </span>
    </h3>
  </a>
  <a
    className="small-movie-card catalog__movies-card small-movie-card__link"
    href="/films/0"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <span
        className="small-movie-card__link"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </span>
    </h3>
  </a>
  <a
    className="small-movie-card catalog__movies-card small-movie-card__link"
    href="/films/0"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <span
        className="small-movie-card__link"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </span>
    </h3>
  </a>
  <a
    className="small-movie-card catalog__movies-card small-movie-card__link"
    href="/films/0"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <span
        className="small-movie-card__link"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </span>
    </h3>
  </a>
  <a
    className="small-movie-card catalog__movies-card small-movie-card__link"
    href="/films/0"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <span
        className="small-movie-card__link"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </span>
    </h3>
  </a>
  <a
    className="small-movie-card catalog__movies-card small-movie-card__link"
    href="/films/0"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <span
        className="small-movie-card__link"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </span>
    </h3>
  </a>
  <a
    className="small-movie-card catalog__movies-card small-movie-card__link"
    href="/films/0"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <span
        className="small-movie-card__link"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </span>
    </h3>
  </a>
</div>
`;
